pennylane/devices/qubit/simulate.py

```diff
@@ -237,7 +257,297 @@ def simulate(
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. If None, a random key will be
+            generated. Only for simulation using JAX.
+        debugger (_Debugger): The debugger to use
+        interface (str): The machine learning interface to create the initial state with
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+        dict: The mid-circuit measurement results of the simulation
+    """
+    state, is_state_batched, mcm_dict = get_final_state(
+        circuit, debugger=debugger, interface=interface
+    )
+    return (
+        measure_final_state(circuit, state, is_state_batched, rng=rng, prng_key=prng_key),
+        mcm_dict,
+    )
+
+
+def accumulate_native_mcm(circuit: qml.tape.QuantumScript, all_shot_meas, one_shot_meas):
+    """Incorporates new measurements in current measurement sequence.
+
+    Args:
+        circuit (QuantumTape): A one-shot (auxiliary) QuantumScript
+        all_shot_meas (Sequence[Any]): List of accumulated measurement results
+        one_shot_meas (Sequence[Any]): List of measurement results
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+    """
+    new_shot_meas = [None] * len(circuit.measurements)
+    if not isinstance(all_shot_meas, (list, tuple)):
+        return accumulate_native_mcm(circuit, [all_shot_meas], one_shot_meas)
+    if not isinstance(one_shot_meas, (list, tuple)):
+        return accumulate_native_mcm(circuit, all_shot_meas, [one_shot_meas])
+    for i, m in enumerate(circuit.measurements):
+        if isinstance(m, CountsMP):
+            tmp = Counter(all_shot_meas[i])
+            tmp.update(Counter(one_shot_meas[i]))
+            new_shot_meas[i] = tmp
+        elif isinstance(m, SampleMP):
+            if not isinstance(all_shot_meas[i], (list, tuple)):
+                new_shot_meas[i] = [all_shot_meas[i]]
+            else:
+                new_shot_meas[i] = all_shot_meas[i]
+            new_shot_meas[i].append(one_shot_meas[i])
+        else:
+            new_shot_meas[i] = all_shot_meas[i] + one_shot_meas[i]
+    return new_shot_meas
+
+
+def has_mid_circuit_measurements(
+    circuit: qml.tape.QuantumScript,
+):
+    """Returns True if the circuit contains a MidMeasureMP object and False otherwise.
+
+    Args:
+        circuit (QuantumTape): A QuantumScript
+
+    Returns:
+        bool: Whether the circuit contains a MidMeasureMP object
+    """
+    return any(isinstance(op, MidMeasureMP) for op in circuit._ops)
```

Comment:
```suggestion
    return any(isinstance(op, MidMeasureMP) for op in circuit.operations)
```

Any reason to access the private version instead?